osu.Game/Screens/BackgroundScreen.cs

```diff
@@ -37,10 +37,17 @@ public override void Push(Screen screen)
             }
 
             // Make sure the in-progress loading is complete before pushing the screen.
-            while (screen.LoadState < LoadState.Ready)
+            while (screen.LoadState != LoadState.Ready)
                 Thread.Sleep(1);
 
-            base.Push(screen);
+            try
+            {
+                base.Push(screen);
+            }
+            catch (InvalidOperationException)
```

Comment:
I really don't like this being a catch-all. What if you try to push and one of the other 4 conditions trigger? The user gets no feedback.

Reply:
Are you okay if it's a specific typed exception?

Reply:
Ah since this is in BackgroundScreen it's not as big of an issue...

Reply:
This is probably going to need updating to match the framework exceptions (+ a nuget update).